exchanges/bitmex/bitmex_wrapper.go

```diff
@@ -248,71 +211,81 @@ func (b *Bitmex) UpdateTradablePairs(forceUpdate bool) error {
 		return err
 	}
 
-	var assetPairs []string
-	for x := range b.CurrencyPairs.AssetTypes {
-		switch b.CurrencyPairs.AssetTypes[x] {
-		case asset.PerpetualContract:
-			for y := range pairs {
-				if strings.Contains(pairs[y], "USD") {
-					assetPairs = append(assetPairs, pairs[y])
-				}
-			}
-		case asset.Futures:
-			for y := range pairs {
-				if strings.Contains(pairs[y], "20") {
-					assetPairs = append(assetPairs, pairs[y])
-				}
-			}
-		case asset.DownsideProfitContract:
-			for y := range pairs {
-				if strings.Contains(pairs[y], "_D") {
-					assetPairs = append(assetPairs, pairs[y])
-				}
-			}
-		case asset.UpsideProfitContract:
-			for y := range pairs {
-				if strings.Contains(pairs[y], "_U") {
-					assetPairs = append(assetPairs, pairs[y])
-				}
-			}
+	// Zerovalue current list which will remove old asset pairs when contract
+	// types expire or become obsolete
+	var assetPairs = map[asset.Item][]string{
+		asset.Index:                  {},
+		asset.PerpetualContract:      {},
+		asset.UpsideProfitContract:   {},
+		asset.DownsideProfitContract: {},
+		asset.Futures:                {},
+	}
+
+	for x := range pairs {
+		if strings.Contains(pairs[x], ".") {
+			assetPairs[asset.Index] = append(assetPairs[asset.Index], pairs[x])
+			continue
+		}
+
+		if strings.Contains(pairs[x], "USD") {
+			assetPairs[asset.PerpetualContract] = append(assetPairs[asset.PerpetualContract],
+				pairs[x])
+			continue
+		}
+
+		assetPairs[asset.Futures] = append(assetPairs[asset.Futures], pairs[x])
+	}
+
+	for asset, values := range assetPairs {
```

Comment:
When dealing with a `map` which is what `assetPairs` is, it can be beneficial to use `for asset, values := range assetPairs` as it actually performs better than `for asset := range assetPairs {`

THAT BEING SAID. Shzbert, please change the var name `asset` to something else as it clashes with the `asset` package